training/distributed_training/pytorch/data_parallel/maskrcnn/train_pytorch_smdataparallel_maskrcnn.py

```diff
@@ -1,180 +1,150 @@
+
 import torch
 from maskrcnn_benchmark.config import cfg
 from maskrcnn_benchmark.data import make_data_loader
+from maskrcnn_benchmark.solver import make_lr_scheduler
+from maskrcnn_benchmark.solver import make_optimizer
 from maskrcnn_benchmark.engine.inference import inference
-from maskrcnn_benchmark.engine.tester import test
 from maskrcnn_benchmark.engine.trainer import do_train
 from maskrcnn_benchmark.modeling.detector import build_detection_model
-from maskrcnn_benchmark.solver import make_lr_scheduler, make_optimizer
 from maskrcnn_benchmark.utils.checkpoint import DetectronCheckpointer
 from maskrcnn_benchmark.utils.collect_env import collect_env_info
-from maskrcnn_benchmark.utils.comm import get_rank, is_main_process, synchronize
+from maskrcnn_benchmark.utils.comm import synchronize, get_rank, is_main_process
 from maskrcnn_benchmark.utils.imports import import_file
 from maskrcnn_benchmark.utils.logger import setup_logger
 from maskrcnn_benchmark.utils.miscellaneous import mkdir
-
-# Import SMDataParallel modules for PyTorch.
-from smdistributed.dataparallel.torch.parallel.distributed import DistributedDataParallel as DDP
-
-dist.init_process_group()
-
-# SMDataParallel: Initialize
-if not dist.is_initialized():
-    dist.init_process_group()
-
-
-# See if we can use apex.DistributedDataParallel instead of the torch default,
-# and enable mixed-precision via apex.amp
+from maskrcnn_benchmark.engine.tester import test
+from maskrcnn_benchmark.utils.logger import format_step
+#from dllogger import Logger, StdOutBackend, JSONStreamBackend, Verbosity
+#import dllogger as DLLogger
+import dllogger
+from maskrcnn_benchmark.utils.logger import format_step
+import smdistributed.dataparallel.torch.torch_smddp
+
+# See if we can use apex.DistributedDataParallel instead of the torch default
 try:
-    from apex import amp
-
-    use_amp = True
+    from apex.parallel import DistributedDataParallel as DDP
+    use_apex_ddp = True
 except ImportError:
-    print("Use APEX for multi-precision via apex.amp")
-    use_amp = False
-# try:
-#     from apex.parallel import DistributedDataParallel as DDP
-#     use_apex_ddp = True
-# except ImportError:
-#     print('Use APEX for better performance')
-use_apex_ddp = False
-
+    print('Use APEX for better performance')
+    use_apex_ddp = False
 
 def test_and_exchange_map(tester, model, distributed):
     results = tester(model=model, distributed=distributed)
 
     # main process only
-    # if is_main_process():
-    if dist.get_rank() == 0:
+    if is_main_process():
         # Note: one indirection due to possibility of multiple test datasets, we only care about the first
         #       tester returns (parsed results, raw results). In our case, don't care about the latter
         map_results, raw_results = results[0]
-        bbox_map = map_results.results["bbox"]["AP"]
-        segm_map = map_results.results["segm"]["AP"]
+        bbox_map = map_results.results["bbox"]['AP']
```

Comment:
Style nit: consistently use either single or double quotes.

Reply:
This is also consistent with [upstream script ](https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py#L57)from NVIDIA repo

Reply:
fixed along with black command